packages/blocks/src/api/raw-handling/paste-handler.js

```diff
@@ -191,6 +193,7 @@ export function pasteHandler( {
 			const filters = [
 				googleDocsUIDRemover,
 				msListConverter,
+				nestedListedConverter,
```

Comment:
Hm, this shouldn't be done for all lists. Ideally, this is added to phrasing content cleanup so that it runs when pasting inline HTML, but also when pasting blocks where a table or figure captions contains lists.

Reply:
Maybe it should be part of `removeInvalidHTML`? Before removing an invalid list, convert it to plain text. We already have some fixing logic there that inserts line breaks when it encounters a block level element (see bottom of `cleanNodeList`).

Reply:
Other than this, code looks good.

Reply:
I've started reworking it so it's compatible with `cleanNodeList `. I can maybe get a PR delivered tomorrow on the train; otherwise, it will be after the meet-up. I will include some integration tests.